app/models/post.rb

```diff
@@ -0,0 +1,21 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: :post_id
+  has_many :likes, foreign_key: :post_id
+
+  before_save :update_posts_counter
+
+  def update_posts_counter
+    count = if author.posts_counter.nil?
+              0
+            else
+              author.posts_counter
+            end
+    author.posts_counter = count + 1
+    author.save
+  end
```

Comment:
- [ ] [OPTIONAL] You can make your code shorter by using something like this 👇🏼 
```ruby
def update_post_counter
  author.increment!(:posts_counter)
end
```

Reply:
Thanks, @Ismailco